collect_app/src/main/java/org/odk/collect/android/tasks/DiskSyncTask.java

```diff
@@ -136,8 +136,11 @@ protected String doInBackground(Void... params) {
                                 uriToUpdate.add(new UriFile(updateUri, sqlFile));
                             }
                         } else {
+                            // remove it from the database also
+                            mFormsDao.deleteFormsForFormId(mCursor.getString(
```

Comment:
@nribeka  Please have a look at this, if [this](https://github.com/opendatakit/collect/pull/578/commits/f940550eabc6073a0bbe50af8cbdf45de888c4e2) works as you described.